pkg/handler/data_recorder_pubsub.go

```diff
@@ -0,0 +1,99 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/swagger_gen/models"
+	"google.golang.org/api/option"
+
+	"github.com/sirupsen/logrus"
+)
+
+type pubsubRecorder struct {
+	enabled  bool
+	producer *pubsub.Client
+	topic    *pubsub.Topic
+}
+
+var (
+	pubsubClient = func() (*pubsub.Client, error) {
+		return pubsub.NewClient(
+			context.Background(),
+			config.Config.RecorderPubsubProjectID,
+			option.WithCredentialsFile(config.Config.RecorderPubsubKeyFile),
+		)
+	}
+)
+
+// NewPubsubRecorder creates a new Pubsub recorder
+var NewPubsubRecorder = func() DataRecorder {
+	client, err := pubsubClient()
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Fatal("error getting pubsub client")
+	}
+
+	return &pubsubRecorder{
+		producer: client,
+		topic:    client.Topic(config.Config.RecorderPubsubTopicName),
+		enabled:  config.Config.RecorderEnabled,
+	}
+}
+
+func (p *pubsubRecorder) AsyncRecord(r *models.EvalResult) {
+	if !p.enabled {
+		return
+	}
+
+	pr := &pubsubEvalResult{
+		EvalResult: r,
+	}
+
+	payload, err := pr.Payload()
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error marshaling")
```

Comment:
Two things:

1) It would be good if these errors (when generating payload and when encoding the message frame) were distinct from one another. Maybe change the error message on this one to "error marshaling payload" and "error marshaling message frame" on the other? Or something to that effect.

2) In both these cases (```encode()``` and ```Payload()```), during the marshaling step the returned values are ```([]byte, error)```. If an error is returned, that means that the payload is an empty byte slice. So in effect what will happen is that a blank message will be written to pubsub. 

In the rare even that this occurs, would it make more sense to return after logging the error? So in each case, have something like:
```go
if err != nil {
  logrus.WithField...
  return
}
```

Reply:
Yep, I do the same on our services... I am trying to comply with your style guide